common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -150,33 +162,16 @@ public Random current() {
             };
         }
 
-        // Here is how the system property is used:
-        //
-        // * <  0  - Don't use cleaner, and inherit max direct memory from java. In this case the
-        //           "practical max direct memory" would be 2 * max memory as defined by the JDK.
-        // * == 0  - Use cleaner, Netty will not enforce max memory, and instead will defer to JDK.
-        // * >  0  - Don't use cleaner. This will limit Netty's total direct memory
-        //           (note: that JDK's direct memory limit is independent of this).
-        long maxDirectMemory = SystemPropertyUtil.getLong("io.netty.maxDirectMemory", -1);
-
-        if (maxDirectMemory == 0 || !hasUnsafe() || !PlatformDependent0.hasDirectBufferNoCleanerConstructor()) {
-            USE_DIRECT_BUFFER_NO_CLEANER = false;
-            DIRECT_MEMORY_COUNTER = null;
-        } else {
-            USE_DIRECT_BUFFER_NO_CLEANER = true;
-            if (maxDirectMemory < 0) {
-                maxDirectMemory = MAX_DIRECT_MEMORY;
-                if (maxDirectMemory <= 0) {
-                    DIRECT_MEMORY_COUNTER = null;
-                } else {
-                    DIRECT_MEMORY_COUNTER = new AtomicLong();
-                }
-            } else {
-                DIRECT_MEMORY_COUNTER = new AtomicLong();
-            }
+        /*
+         * We do not want to log this message if unsafe is explicitly disabled. Do not remove the explicit no unsafe
+         * guard.
+         */
+        if (!hasUnsafe() && !isAndroid() && !PlatformDependent0.isExplicitNoUnsafe()) {
```

Comment:
```suggestion
        if (!hasUnsafe() && !PlatformDependent0.isExplicitNoUnsafe()) {
```
Is there a plan to release it for Android?

Reply:
This is to keep the original behavior of netty and with our changes.